src/SimulationRunner.cc

```diff
@@ -124,20 +124,23 @@ SimulationRunner::SimulationRunner(const sdf::World *_world,
   // If the configuration is invalid, then networkMgr will be `nullptr`.
   if (_config.UseDistributedSimulation())
   {
-    if (_config.NetworkRole().empty())
+    if (_config.NetworkRole() == "primary" ||
```

Comment:
I changed this bit because at some point there is a partial/incomplete support for ReadOnly network types. However, trying to use those interferes with the rest of the network code that closes the simulation after a node is removed.